apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -2881,6 +2913,9 @@ impl FetchSelectionSet {
 
     fn add_selections(&mut self, selection_set: &Arc<SelectionSet>) -> Result<(), FederationError> {
         Arc::make_mut(&mut self.selection_set).add_selection_set(selection_set)?;
+        // TODO: when calling this multiple times, maybe only re-compute conditions at the end?
+        // Or make it lazily-initialized and computed on demand?
```

Comment:
Is there a ticket for this? I know there is a similar note in `add_at_path`. If so, maybe we should make a note on that ticket.

Reply:
Unsure if there is one, I just copied over same comment so when folks look at the TODOs they can tackle both at the same time. The only place we are reading it is in `process_node` so I guess there is not much benefit of storing it on the `FetchSelectionSet` as we can always just compute it directly from its selection set (this is what we do in this assignment). Which in turn brings a question do we even need a `FetchSelectionSet` wrapper in the first place (as it would only store selection set).